sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/trace/zpages/ZPageServer.java

```diff
@@ -195,7 +196,17 @@ private void stop() {
    * @return new SdkTracerProvider
    */
   public SdkTracerProvider buildSdkTracerProvider() {
-    return SdkTracerProvider.builder()
+    return buildSdkTracerProvider(SdkTracerProvider.builder());
+  }
+
+  /**
+   * Convenience method to return a new SdkTracerProvider that has been configured with our ZPage
+   * specific span processor, sampler, and limits.
+   *
+   * @return new SdkTracerProvider
+   */
+  public SdkTracerProvider buildSdkTracerProvider(SdkTracerProviderBuilder builder) {
```

Comment:
I would have this return void, but don't feel strongly since its a convenience method and is experimental.

Reply:
Cool. Yeah I totally see where you're coming from, but pure side-effect methods also give me the creeps. :)